libraries/lib-wave-track/SequenceSampleCache.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
```

Comment:
Add banner comments

Reply:
... or don't make a separate file?

You made this cache separate from Sequence and used a std::function callback -- were you thinking of avoiding dependency cycle?  Yes, it avoids the cycle between two translation units.

But now I think it could be quite proper for Sequence and its cache to cooperate intimately, each class using the other's interface directly, and one a friend of the other -- and so, put them all just in Sequence.*

Avoidance of dependency cycles is desirable in the graph if `#include`s relating translation units.  But it's not necessary at the level of closely cooperating classes, like containers and their iterators.

Reply:
And the next thing is we spend time splitting classes into different headers/translation units :-)

Reply:
> And the next thing is we spend time splitting classes into different headers/translation units :-)

@crsib We split lots of old legacy code that combined UI and internals carelessly and was generally written without regard to proper dependency cycle management.  This case isn't like that.

But are you thinking of some generalized abstract, polymorphic Sequence?  I think you hinted at that and I don't undertand you yet.